scripts/js/playlists.js

```diff
@@ -110,37 +98,27 @@ else if (type == 'm3u')
         }
         
         line = readln();
-    }
-    while (line);
-}
-else if (type == 'pls')
-{
-    var title = null;
-    var file = null;
-    var lastId = -1;
-    var line = readln();
-    do
-    {
-        if (line.match(/^\[playlist\]$/i))
-        {
+    } while (line);
+} else if (type === 'pls') {
+    title = null;
+    file = null;
+    lastId = -1;
+    line = readln();
+    do {
+        if (line.match(/^\[playlist\]$/i)) {
             // It seems to be a correct playlist, but we will try to parse it
             // anyway even if this header is missing, so do nothing.
-        }
-        else if (line.match(/^NumberOfEntries=(\d+)$/i))
-        {
+        } else if (line.match(/^NumberOfEntries=(\d+)$/i)) {
             // var numEntries = RegExp.$1;
-        }
-        else if (line.match(/^Version=(\d+)$/i))
-        {
+        } else if (line.match(/^Version=(\d+)$/i)) {
             // var plsVersion =  RegExp.$1;
-        }
-        else if (line.match(/^File\s*(\d+)\s*=\s*(\S.+)$/i))
-        {
-            var thisFile = RegExp.$2;
-            var id = parseInt(RegExp.$1, 10);
-            if (lastId == -1)
+        } else if (line.match(/^File\s*(\d+)\s*=\s*(\S.+)$/i)) {
+            matches = line.match(/^File\s*(\d+)\s*=\s*(\S.+)$/i);
```

Comment:
Here is an example where the script used to use `RegExp.$2`.  I could not find a way to make this work.  Fortunately, getting the returned matches from the function itself works fine....somewhat duplicative...but OK.